unlock-js/src/v0/createLock.js

```diff
@@ -14,15 +14,10 @@ export default async function(lock, callback) {
   if (maxNumberOfKeys === UNLIMITED_KEYS_COUNT) {
     maxNumberOfKeys = ETHERS_MAX_UINT
   }
-  const transactionPromise = unlockContract.functions[
-    'createLock(uint256,uint256,uint256)'
-  ](
+  const transactionPromise = unlockContract.createLock(
     lock.expirationDuration,
     ethersUtils.toWei(lock.keyPrice, 'ether'),
-    maxNumberOfKeys,
-    {
-      gasLimit: GAS_AMOUNTS.createLock,
```

Comment:
ethers is actually smart enough to extract the expected gas price for a transaction, so there is no need to pass one.